packages/mdc-slider/foundation.js

```diff
@@ -0,0 +1,268 @@
+      deregisterBodyEventHandler: (/* type: string, handler: EventListener */) => {},
+      registerResizeHandler: (/* handler: EventListener */) => {},
+      deregisterResizeHandler: (/* handler: EventListener */) => {},
+      notifyInput: () => {},
+      notifyChange: () => {},
+      setThumbStyleProperty: (/* propertyName: string, value: string */) => {},
+      setTrackFillStyleProperty: (/* propertyName: string, value: string */) => {},
+    });
+  }
+
+  /**
+   * Creates a new instance of MDCSliderFoundation
+   * @param {?MDCSliderAdapter} adapter
+   */
+  constructor(adapter) {
+    super(Object.assign(MDCSliderFoundation.defaultAdapter, adapter));
+    /** @private {?ClientRect} */
+    this.rect_ = null;
+    /** @private {boolean} */
+    this.active_ = false;
+    /** @private {number} */
+    this.min_ = 0;
+    /** @private {number} */
+    this.max_ = 100;
+    /** @private {number} */
+    this.value_ = 0;
+    /** @private {boolean} */
+    this.handlingThumbTargetEvt_ = false;
+    /** @private {function(): undefined} */
+    this.thumbPointerHandler_ = () => {
+      this.handlingThumbTargetEvt_ = true;
+    };
+    /** @private {function(!Event): undefined} */
+    this.eventStartHandler_ = (evt) => this.handleDown_(evt);
+    /** @private {function(): undefined} */
+    this.resizeHandler_ = () => this.layout();
+  }
+
+  init() {
+    DOWN_EVENTS.forEach((evtName) => this.adapter_.registerEventHandler(evtName, this.eventStartHandler_));
+    DOWN_EVENTS.forEach((evtName) => {
+      this.adapter_.registerThumbEventHandler(evtName, this.thumbPointerHandler_);
+    });
+    this.adapter_.registerResizeHandler(this.resizeHandler_);
+    this.layout();
+  }
+
+  destroy() {
+    DOWN_EVENTS.forEach((evtName) => {
+      this.adapter_.deregisterEventHandler(evtName, this.eventStartHandler_);
+    });
+    DOWN_EVENTS.forEach((evtName) => {
+      this.adapter_.deregisterThumbEventHandler(evtName, this.thumbPointerHandler_);
+    });
+    this.adapter_.deregisterResizeHandler(this.resizeHandler_);
+  }
+
+  layout() {
+    this.rect_ = this.adapter_.computeBoundingRect();
+    this.updateUIForCurrentValue_();
+  }
+
+  /** @return {number} */
+  getValue() {
+    return this.value_;
+  }
+
+  /** @param {number} value */
+  setValue(value) {
+    this.setValue_(value);
+  }
+
+  /** @return {number} */
+  getMax() {
+    return this.max_;
+  }
+
+  /** @param {number} max */
+  setMax(max) {
+    if (max < this.min_) {
+      throw new Error('Cannot set max to be less than the slider\'s minimum value');
+    }
+    this.max_ = max;
+    this.setValue_(this.value_);
+    this.adapter_.setAttribute('aria-valuemax', String(this.max_));
+  }
+
+  /** @return {number} */
+  getMin() {
+    return this.min_;
+  }
+
+  /** @param {number} min */
+  setMin(min) {
+    if (min > this.max_) {
+      throw new Error('Cannot set min to be greater than the slider\'s maximum value');
```

Comment:
no-op!

Reply:
I'm tentatively OK with a no-op here, since this coincides with pre-existing issue #2269 which was caused by this. (It's probably feasible to avoid throwing an error in that particular instance somehow, though.)

In general though, throwing errors can be preferable so that a developer can actually identify what went wrong when something unexpected happens (or doesn't happen, in this case). Silent failures are some of the _worst_ things to debug.